lib/editor/tunnel.js

```diff
@@ -202,21 +202,29 @@ class EditorTunnel {
             })
         }
         socket.on('close', async (code, reason) => {
+            if (Buffer.isBuffer(reason)) {
+                reason = reason.toString()
+            }
             // The socket connection to the platform has closed
             info(`Editor tunnel closed code=${code} reason=${reason}`)
             socket.removeAllListeners()
             this.socket = null
             clearTimeout(this.reconnectTimeout)
-
-            // Assume we need to be reconnecting. If this close is due
-            // to a request from the platform to turn off editor access,
-            // .close will get called
-            const reconnectDelay = this.reconnectDelay
-            // Bump the delay for next time... 500ms, 1.5s, 4.5s, 10s 10s 10s...
-            this.reconnectDelay = Math.min(this.reconnectDelay * 3, 10000)
-            this.reconnectTimeout = setTimeout(() => {
-                this.connect()
-            }, reconnectDelay)
+            if (reason !== 'No tunnel') {
```

Comment:
What generates this text Nick?

Just thinking, if it is our code in the platform (or even if it is generated in socket package), a change of text would bypass this check.

Should/could we use a well defined code instead of reason text?

Reply:
This text does come from the platform and is set by us. I agree it's a little brittle. Will consider a change of code - need to look at what the ws spec allows.

Reply:
Have updated the PR to look for either:

 - `1008/No tunnel` - for FF 1.11 and earlier
 - `4004` - for FF 1.12

We are using `100*` codes in multiple places where we should not be - they are all protocol specific reserved codes we shouldn't be overloading. I have made some minor changes in the platform PR to use 400* range codes which are the ones reserved for application specific errors. Will need to do a broader tidy up.